src/source/Common/EcsCredentialProvider.c

```diff
@@ -255,7 +255,7 @@ STATUS getHostPort(PCHAR pUrl, PCHAR* ppStart, PCHAR* ppEnd)
     CHK(pStart != NULL, STATUS_INVALID_ARG);
 
     // Advance the pStart past the delimiter
-    pStart += (ARRAY_SIZE(PORT_DELIMITER_STRING) - 1);
+    pStart += STRLEN(PORT_DELIMITER_STRING);
```

Comment:
Could you elaborate what you mean by "incorrect"? ARRAY_SIZE macro should return for a string literal ":"

Reply:
Oh, I see... You defined the literal like this:

`#define PORT_DELIMITER_STRING (PCHAR) ":"`

This means that it's a pointer and not the actual string.

Try instead

`#define PORT_DELIMITER_STRING ":"`

And, whenever you use this as a non-const then cast to PCHAR (for C++ compiler warning)

Reply:
@ycyang1229 Yes please change this so we can keep ARRAY_SIZE

Reply:
@MushMal thanks. :) Changed it back. I wonder what is the code convention of this since there are different macros for literals. I thought I need to follow this line, but it seems I took the wrong line. 
/**
 * Schema delimiter string
 */
#define SCHEMA_DELIMITER_STRING (PCHAR) "://"
@hassanctech done